spec.md

```diff
@@ -1663,6 +1717,514 @@ message GetCapacityResponse {
+
+#### `ControllerGetVolumeGroupSnapshot`
+
+**ALPHA FEATURE**
+
+```protobuf
+message ControllerGetVolumeGroupSnapshotRequest {
+  option (alpha_message) = true;
+
+  // The ID of the group snapshot to fetch current group snapshot
+  // information for.
+  // This field is REQUIRED.
+  string group_snapshot_id = 1;
+
+  // Secrets required by plugin to complete
+  // ControllerGetVolumeGroupSnapshot request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> secrets = 2 [(csi_secret) = true];
+}
+
+message ControllerGetVolumeGroupSnapshotResponse {
+  option (alpha_message) = true;
+
+  // This field is REQUIRED
+  VolumeGroupSnapshot group_snapshot = 1;
+}
+```
+
+##### ControllerGetVolumeGroupSnapshot Errors
+
+If the plugin is unable to complete the ControllerGetVolumeGroupSnapshot call successfully, it MUST return a non-ok gRPC code in the gRPC status.
+If the conditions defined below are encountered, the plugin MUST return the specified gRPC error code.
+The CO MUST implement the specified error recovery behavior when it encounters the gRPC error code.
+
+| Condition | gRPC Code | Description | Recovery Behavior |
+|-----------|-----------|-------------|-------------------|
+| Volume group snapshot does not exist | 5 NOT_FOUND | Indicates that a volume group snapshot corresponding to the specified `volume_group_snapshot_id` does not exist. | Caller MUST verify that the `volume_group_snapshot_id` is correct and that the volume group snapshot is accessible and has not been deleted before retrying with exponential back off. |
+
+#### `ListVolumeGroupSnapshots`
+
+**ALPHA FEATURE**
+
+```protobuf
+message ListVolumeGroupSnapshotsRequest {
+  option (alpha_message) = true;
+
+  // If specified (non-zero value), the Plugin MUST NOT return more
+  // entries than this number in the response. If the actual number of
+  // entries is more than this number, the Plugin MUST set `next_token`
+  // in the response which can be used to get the next page of entries
+  // in the subsequent `ListVolumeGroupSnapshots` call.
+  // This field is OPTIONAL. If not specified (zero value), it means
+  // there is no restriction on the number of entries that can be
+  // returned. The value of this field MUST NOT be negative.
+  int32 max_entries = 1;
+
+  // A token to specify where to start paginating. Set this field to
+  // `next_token` returned by a previous `ListVolumeGroupSnapshots`
+  // call to get the next page of entries. This field is OPTIONAL.
+  // An empty string is equal to an unspecified field value.
+  string starting_token = 2;
+
+  // Identity information for the source volume group.
+  // This field is OPTIONAL.
+  // It can be used to list group snapshots by volume group.
```

Comment:
if some backends cannot create snapshots w/o a list of volumes in the group, presumably the "list" API has the same weakness? if not, how does that make sense?

Reply:
This is an optional capability.  A backend can opt out if it can't support it.